lightkurve/collection.py

```diff
@@ -0,0 +1,101 @@
+from abc import abstractmethod, ABCMeta
+
+__all__ = ['Collection']
+
+class Collection:
+    """Abstract Base Class for LightCurveCollections, TPFCollections
+
+    Attributes
+    ----------
+    data: array
+        List of data objects.
+    k_id: dictionary
+        Mapping keplerid to index in self.data
+    """
+    __metaclass__ = ABCMeta #Needs to be set for Python 2.x to work properly
+    def __init__(self, data):
+        self.data = data
+        self.k_id = self.assign_hash_values()
+
+    def assign_hash_values(self):
+        """
+        Assigns the keplerid to indexes in self.data
+
+        Parameters:
+        -----------
+        None
+
+        Returns
+        -------
+        result: Dictionary
+            With keys of keplerid (int) and
+            values of indexes (int) in the data array.
+        """
+        result = {}
+        for idx,obj in enumerate(self.data):
+            try:
+                result[obj.keplerid] = idx
+            except AttributeError:
+                print("Object "+ str(idx) + " has no keplerid")
+        return result
+
+    def __len__(self):
+        return len(self.data)
+
+    def __getitem__(self, index):
+        """Called when indexing into LCC or such as when looping.
```

Comment:
Let's make this docstring more clear, e.g. `Called when accessing an element in the collection by index or target id.`